File: examples/python/ok_basics.py
# Functions, assignment, arithmetic, string concatenation and a
# for-each loop over a list. Types are inferred throughout.

def add(a, b):
    return a + b

def double(n):
    return n * 2

total = 0
for i in [1, 2, 3, 4, 5]:
    total = total + add(i, double(i))

label = "total = " + "done"

File: examples/lua/ok_basics.lua
-- Functions, locals, arithmetic, string concatenation and a numeric
-- for loop. Everything here is inferred — no annotations.

local function add(a, b)
  return a + b
end

local function double(n)
  return n * 2
end

local sum = 0
for i = 1, 10 do
  sum = sum + add(i, double(i))
end

-- `..` is string concatenation; it lowers to `+` (the Plus type class
-- on strings), so mixing a string with a number is still rejected.
local label = "sum = " .. "done"
